current/advanced/block-import.md

```diff
@@ -0,0 +1,90 @@
+them, and if they are valid, importing them into the node's state. The most fundamental piece of
+information that the import queue processes is blocks themselves, but it is also responsible for
+importing consensus-related messages such as Justifications and, in light clients, finality proofs.
+
+The import queue collects incoming blocks from the network and stores them in a pool. The blocks are
+later checked for validity and discarded if they are not valid. Blocks that are valid are then
+imported into the node's local state.
+
+The import queue is codified abstractly in Substrate by means of the
+[`ImportQueue` trait](https://substrate.dev/rustdocs/v2.0.0-alpha.8/sp_consensus/import_queue/trait.ImportQueue.html).
+The use of a trait allows each consensus engine to provide its own specialized implementation of the
+Import Queue which may take advantage of optimization opportunities such as verifying multiple
+blocks in parallel as they come in across the network.
+
+The import queue also provides some hooks via the
+[`Link` trait](https://crates.parity.io/sp_consensus/import_queue/trait.Link.html) that can be used
+to follow its progress. A typical consumer of these hooks is the sync module which uses data from
+the import queue to adjust its sync algorithm.
+
+## The Basic Queue
+
+Substrate provides a default in-memory implementation of the `ImportQueue` known as the
+[`BasicQueue`](https://substrate.dev/rustdocs/v2.0.0-alpha.8/sp_consensus/import_queue/struct.BasicQueue.html).
+The `BasicQueue` does not do any kind of optimization, rather it performs the verification and
+import steps sequentially. It does, however, abstract the notion of verification through the use of
+the
+[`Verifier`](https://substrate.dev/rustdocs/v2.0.0-alpha.8/sp_consensus/import_queue/trait.Verifier.html)
+trait.
+
+Any consensus engine that relies on the `BasicQueue` must implement the `Verifier` trait. The
+`Verifier` is typically responsible for tasks such as checking
+[inherent data](https://www.substrate.io/kb/learn-substrate/extrinsics#inherents), and ensuring that
+the block is signed by the appropriate authority.
+
+## The Block Import Trait
+
+When the import queue is ready to import a block, it passes the block in question to a method
+provided by the
+[`BlockImport` trait](https://substrate.dev/rustdocs/v2.0.0-alpha.8/sp_consensus/block_import/trait.BlockImport.html).
+This `BlockImport` trait provides the behavior of importing a block into the node's local state
+database.
+
+One implementor of the `BlockImport` trait that is used in every Substrate node is the
+[`Client`](https://substrate.dev/rustdocs/v2.0.0-alpha.8/sc_service/client/index.html) which
+contains the node's entire block database. When a block is imported into the client it is added to
+the main database of blocks that the node knows about.
+
+## The Block Import Pipeline
+
+In the simplest cases, such as
```

Comment:
Why is manual seal the simplest case? I almost wonder if it may not be better to leave out that example. Do you think it would be sufficient to just say "In the simplest cases, blocks are imported directly..."?